File: frontend/src/utils/encryption.ts
// 简单的Base64编码加密（用于演示，生产环境应使用更强的加密）
/**
 * 前端加密工具类
 * 用于敏感数据的加密和解密
 */
class EncryptionManager {
  private secretKey: string;

  constructor() {
    // 使用固定的密钥，在生产环境中应该从环境变量获取
    this.secretKey = 'your-secret-key-for-frontend-encryption-2024';
  }

  /**
   * 简单加密字符串（Base64 + 简单混淆）
   */
  encrypt(data: string): string {
    if (!data) return data;

    try {
      // 简单的加密：Base64编码 + 密钥混淆
      const combined = data + '|' + this.secretKey.substring(0, 8);
      const encoded = btoa(unescape(encodeURIComponent(combined)));
      return encoded;
    } catch (error) {
      console.error('加密失败:', error);
      return data;
    }
  }

  /**
   * 简单解密字符串
   */
  decrypt(encryptedData: string): string {
    if (!encryptedData) return encryptedData;

    try {
      const decoded = decodeURIComponent(escape(atob(encryptedData)));
      const parts = decoded.split('|');
      if (parts.length === 2 && parts[1] === this.secretKey.substring(0, 8)) {
        return parts[0];
      }
      return encryptedData;
    } catch (error) {
      console.error('解密失败:', error);
      return encryptedData;
    }
  }

  /**
   * 加密敏感数据对象
   */
  encryptSensitiveData(data: any): any {
    const sensitiveFields = ['password', 'phone', 'email', 'old_password', 'new_password'];
    const encryptedData = { ...data };

    sensitiveFields.forEach(field => {
      if (encryptedData[field]) {
        encryptedData[field] = this.encrypt(String(encryptedData[field]));
      }
    });

    return encryptedData;
  }

  /**
   * 解密敏感数据对象
   */
  decryptSensitiveData(data: any): any {
    const sensitiveFields = ['phone', 'email']; // 密码不需要解密显示
    const decryptedData = { ...data };

    sensitiveFields.forEach(field => {
      if (decryptedData[field]) {
        decryptedData[field] = this.decrypt(String(decryptedData[field]));
      }
    });

    return decryptedData;
  }
}

// 创建全局实例
const encryptionManager = new EncryptionManager();

/**
 * 加密密码
 */
export const encryptPassword = (password: string): string => {
  return encryptionManager.encrypt(password);
};

/**
 * 加密电话号码
 */
export const encryptPhone = (phone: string): string => {
  return encryptionManager.encrypt(phone);
};

/**
 * 解密电话号码
 */
export const decryptPhone = (encryptedPhone: string): string => {
  return encryptionManager.decrypt(encryptedPhone);
};

/**
 * 加密邮箱
 */
export const encryptEmail = (email: string): string => {
  return encryptionManager.encrypt(email);
};

/**
 * 解密邮箱
 */
export const decryptEmail = (encryptedEmail: string): string => {
  return encryptionManager.decrypt(encryptedEmail);
};

/**
 * 加密登录数据
 */
export const encryptLoginData = (loginData: { username: string; password: string }) => {
  return {
    username: loginData.username, // 用户名不加密
    password: encryptPassword(loginData.password)
  };
};

/**
 * 加密用户创建数据
 */
export const encryptUserCreateData = (userData: any) => {
  return encryptionManager.encryptSensitiveData(userData);
};

/**
 * 加密用户更新数据
 */
export const encryptUserUpdateData = (userData: any) => {
  return encryptionManager.encryptSensitiveData(userData);
};

/**
 * 加密密码更新数据
 */
export const encryptPasswordUpdateData = (passwordData: { old_password: string; new_password: string }) => {
  return {
    old_password: encryptPassword(passwordData.old_password),
    new_password: encryptPassword(passwordData.new_password)
  };
};

/**
 * 解密用户数据（用于显示）
 */
export const decryptUserData = (userData: any) => {
  return encryptionManager.decryptSensitiveData(userData);
};

export default encryptionManager;

File: backend/app/core/encryption.py
import base64
import hashlib
from typing import Optional
from .config import settings


class EncryptionManager:
    """加密管理器 - 用于敏感数据的加密和解密，与前端简单加密兼容"""

    def __init__(self):
        self.secret_key = getattr(settings, 'ENCRYPTION_KEY', 'your-secret-key-for-frontend-encryption-2024')

    def encrypt(self, data: str) -> str:
        """简单加密字符串数据，与前端兼容"""
        if not data:
            return data

        try:
            # 简单的加密：Base64编码 + 密钥混淆
            combined = data + '|' + self.secret_key[:8]
            encoded = base64.b64encode(combined.encode('utf-8')).decode('utf-8')
            return encoded

        except Exception as e:
            print(f"加密失败: {e}")
            return data

    def decrypt(self, encrypted_data: str) -> str:
        """简单解密字符串数据，与前端兼容"""
        if not encrypted_data:
            return encrypted_data

        try:
            # 解码base64
            decoded = base64.b64decode(encrypted_data.encode('utf-8')).decode('utf-8')
            parts = decoded.split('|')

            if len(parts) == 2 and parts[1] == self.secret_key[:8]:
                return parts[0]
            return encrypted_data

        except Exception as e:
            print(f"解密失败: {e}")
            return encrypted_data
    
    def encrypt_sensitive_data(self, data: dict) -> dict:
        """加密敏感数据字段"""
        sensitive_fields = ['password', 'phone', 'email']
        encrypted_data = data.copy()
        
        for field in sensitive_fields:
            if field in encrypted_data and encrypted_data[field]:
                encrypted_data[field] = self.encrypt(str(encrypted_data[field]))
        
        return encrypted_data
    
    def decrypt_sensitive_data(self, data: dict) -> dict:
        """解密敏感数据字段"""
        sensitive_fields = ['phone', 'email']  # 密码不需要解密显示
        decrypted_data = data.copy()
        
        for field in sensitive_fields:
            if field in decrypted_data and decrypted_data[field]:
                decrypted_data[field] = self.decrypt(str(decrypted_data[field]))
        
        return decrypted_data


# 全局加密管理器实例
encryption_manager = EncryptionManager()


def encrypt_password(password: str) -> str:
    """加密密码（用于传输）"""
    return encryption_manager.encrypt(password)


def decrypt_password(encrypted_password: str) -> str:
    """解密密码（用于验证）"""
    return encryption_manager.decrypt(encrypted_password)


def encrypt_phone(phone: str) -> str:
    """加密电话号码"""
    return encryption_manager.encrypt(phone) if phone else phone


def decrypt_phone(encrypted_phone: str) -> str:
    """解密电话号码"""
    return encryption_manager.decrypt(encrypted_phone) if encrypted_phone else encrypted_phone


def encrypt_email(email: str) -> str:
    """加密邮箱"""
    return encryption_manager.encrypt(email) if email else email


def decrypt_email(encrypted_email: str) -> str:
    """解密邮箱"""
    return encryption_manager.decrypt(encrypted_email) if encrypted_email else encrypted_email


def hash_for_storage(data: str) -> str:
    """为存储生成哈希值（不可逆）"""
    return hashlib.sha256(data.encode('utf-8')).hexdigest()
